MINIBALL: add CPU virtualization docs

yaoyinnan opened the pull request:
Add description document for CPU virtualization part. Modify the description of CPU virtualization in MiniBall's README.md.

Fixes: https://github.com/openanolis/dragonball-sandbox/issues/187

Signed-off-by: yaoyinnan <yaoyinnan@foxmail.com>

ZizhengBian commented on crates/dbs-miniball/docs/CPUVirtualization.md:
```diff
@@ -0,0 +1,869 @@
+# CPU 虚拟化
+
+## 概述
+在配置 KVM、guest memory 后，开始配置 vCPU。通过 dbs-boot、dbs-arch、kvm-ioctls crates 完成。
```
Pls use English or Chinese&English

studychao replied:
As I have discussed with @yaoyinnan, he should first finish all the virtualization-related documents in his preferred language. And he could translate it to English when he has extra time. The important part of this project is virtualization knowledge instead of language translation.

But @yaoyinnan, could you add an comment/tip here to tell potential readers why there is only Chinese here?

yaoyinnan replied:
Yes, as @studychao said. Since my current main task is to build a simple virtual machine to sort out the virtualization knowledge points, the document is in Chinese. After I have free time, I will sync it for translation.

ZizhengBian commented on crates/dbs-miniball/docs/CPUVirtualization.md:
```diff
@@ -0,0 +1,869 @@
+
+为了获得处理器信息，操作系统有两种获取处理器信息的方式：一种是 Intel 的 Multi Processor Specification（后续简称 MP Spec）约定的方式；另外一种是 ACPI MADT（Multiple APIC Description Table）约定的方式。MP Spec 约定的核心数据结构包括两部分：MP Floating Pointer Structure（后续简称 MPF）和 MP  Configuration Table（后续简称 MP Table）的地址，如下图所示。
+
+![MPtable.png](img/MPTable.png)
```
Redraw the picture by yourself and add not only the picture but the original file is better.

yaoyinnan replied:
OK, I'll revise it later.

yaoyinnan commented:
Since my current main task is to build a simple virtual machine to sort out the virtualization knowledge points, the document is in Chinese. After I have free time, I will sync it for translation.

studychao commented on crates/dbs-miniball/docs/CPUVirtualization.md:
````diff
@@ -0,0 +1,869 @@
+.map_err(Error::MpTable)?;
+```
+
+在 `setup_mptable()` 中，第 5～7 行，首先判断 boot_cpus 是否超过给定的 max_cpus，以及第 8～9 行，判断 max_cpus 是否大于系统支持的最大 CPU 数目（MAX_SUPPORTED_CPUS）。之后，kvmtool 将 MP Table 放置在主板 BIOS 所在的区域（0xF0000 ～ 0xFFFFF），在 BIOS 实际占据地址的末尾处。kvmtool 首先申请了一块内存区，在其中组织 MPF 和 MP Table，然后将组织好的数据结构复制到 Guest 中主板 BIOS 所在的区域。代码如下：
````
kvmtool is another VMM.
In this document, you should use `miniball`

studychao commented:
Hi Yinnan, thanks for your amazing work. 
Overall, LGTM. only one small comment.

yaoyinnan committed:
MINIBALL: add CPU virtualization docs

Add description document for CPU virtualization part. Modify the description of CPU virtualization in MiniBall's README.md.

Fixes: https://github.com/openanolis/dragonball-sandbox/issues/187

Signed-off-by: yaoyinnan <yaoyinnan@foxmail.com>
